pkg/docker/docker.go

```diff
@@ -126,6 +126,62 @@ type RunContainerOptions struct {
 	CapDrop          []string
 }
 
+// asDockerConfig converts a RunContainerOptions into a Config understood by the
+// docker client library.
+func (rco RunContainerOptions) asDockerConfig() docker.Config {
+	return docker.Config{
+		Image: getImageName(rco.Image),
+		User:  rco.User,
+		// TODO why do we call runContainerTar with a "partial"
+		// docker.Config?
+
+		Env: rco.Env,
+
+		// TODO review how callers use these options to pass stdin.
+		// Probably they are not necessary, as they're related to
+		// attaching to the container (that is often unnecessary).
+		OpenStdin: rco.Stdin != nil,
+		StdinOnce: rco.Stdin != nil,
+		// TODO surprinsingly, we do not set AttachStdin...
+
+		AttachStdout: rco.Stdout != nil,
+	}
+}
+
+// asDockerHostConfig converts a RunContainerOptions into a HostConfig
+// understood by the docker client library.
+func (rco RunContainerOptions) asDockerHostConfig() docker.HostConfig {
+	hostConfig := docker.HostConfig{
+		CapDrop: rco.CapDrop,
+
+		PublishAllPorts: rco.TargetImage,
+		// TODO why didn't we always set NetworkMode?!
```

Comment:
The defaults are the zero values of the field types, for strings it's the empty string. We can always override when we're building the struct from scratch. So we're fine here.

Do you know what TargetImage actually means, semantically?

It's used to set `PublishAllPorts`, to terminate early `runContainerTar`, and to call `runContainerDockerRun`...